includes/wcs-compatibility-functions.php

```diff
@@ -570,3 +570,39 @@ function wcs_add_woocommerce_dependent_action( $tag, $function, $woocommerce_ver
 function wcs_is_woocommerce_pre( $version ) {
 	return ! defined( 'WC_VERSION' ) || version_compare( WC_VERSION, $version, '<' );
 }
+
+/**
+ * Checks if the WooCommerce feature is enabled using WC's new FeaturesUtil class.
+ *
+ * @param string $feature_name The name of the WC feature to check if enabled.
+ *
+ * @return bool
+ */
+function wcs_is_wc_feature_enabled( $feature_name ) {
+	$feature_is_enabled = false;
+
+	if ( class_exists( '\Automattic\WooCommerce\Utilities\FeaturesUtil' ) && \Automattic\WooCommerce\Utilities\FeaturesUtil::feature_is_enabled( $feature_name ) ) {
+		$feature_is_enabled = true;
+	}
+
+	return $feature_is_enabled;
+}
+
+/**
+ * Helper function to determine whether custom orders table usage is enabled.
+ *
+ * Custom order table feature can be enabled but the store is still using WP posts as the authoriative source of order data,
+ * therefore this function will only return true if:
+ *  - the HPOS feature is enabled
+ *  - the HPOS tables have been generated
+ *  - HPOS is the authoriative source of order data
+ *
+ * @return bool
+ */
+function wcs_is_custom_order_tables_usage_enabled() {
+	if ( ! class_exists( '\Automattic\WooCommerce\Utilities\OrderUtil' ) || ! wcs_is_wc_feature_enabled( 'custom_order_tables' ) ) {
+		return false;
+	}
+
+	return \Automattic\WooCommerce\Utilities\OrderUtil::custom_orders_table_usage_is_enabled();
```

Comment:
both `wcs_is_wc_feature_enabled( 'custom_order_tables' )` and `OrderUtil::custom_orders_table_usage_is_enabled();` use a simple `get_option()` so not very taxing. 
Will there ever be a case where `wcs_is_wc_feature_enabled( 'custom_order_tables' )` is false while `OrderUtil::custom_orders_table_usage_is_enabled();` is true? I'm wondering if we can remove the `wcs_is_wc_feature_enabled` check.

Reply:
> Will there ever be a case where `wcs_is_wc_feature_enabled( 'custom_order_tables' )` is false while `OrderUtil::custom_orders_table_usage_is_enabled();` is true? 

From my understanding and testing this is possible. If I have the COT tables turned on in settings (https://d.pr/i/QzlhJ9) and then turn off the HPOS feature (https://d.pr/draw/X9ZBnX/), the `woocommerce_custom_orders_table_enabled` option in the database is still set to 'yes'.

So we can't just use `OrderUtil::custom_orders_table_usage_is_enabled()` and have to check both the feature is enabled and the COT tables are in-use.